exam/summer2024/build.py

```diff
@@ -0,0 +1,42 @@
+import os
+import fileinput
+from subprocess import call
+
+os.makedirs('built', exist_ok=True)
+call_pdflatex_l = ['pdflatex', '-synctex=1',
+                   '-interaction=nonstopmode', 'main.tex']
+
+
+def clear_tex_binaries():
+    # clean misc files
+    for file in os.listdir('.'):
+        if file.startswith('main'):
+            if not file.endswith(('.tex', '.pdf')):
+                os.remove(file)
+
+
+# build main pdf
+clear_tex_binaries()
+with fileinput.input('main.tex', inplace=True) as f:
+    for line in f:
+        if 'includeonly{' in line:
+            # comment out includeonly flag
+            print(f'%{line}', end='')
+        #check if the line including '\documentclass' has the parameter 'handout' - if not add it
+        elif ']{beamer}' in line:
+            if 'handout' not in line:
+                print(line.replace(']{beamer}', ', handout]{beamer}'), end='')
+            else:
+                print(line, end='')    
+        else:
```

Comment:
The exams are not not based on latex beamer class but on the article class -> compare examClass.cls. Hence, there is no handout option for the exams.

Instead, there is a "[solution]" option within the exam class which can be utilized to not only print the exam task but also to directly print out the solutions after each tasks. 

Hence, per exam we need to build two pdfs. One just covering the tasks and one include the solutions. This enables students to utilize first the version without solutions for training and then to check their results later.